transaction-coordinator/src/main/java/org/apache/kafka/coordinator/transaction/TransactionState.java

```diff
@@ -0,0 +1,164 @@
+     *             received EndTxnRequest with abort and TransactionV2 enabled => PrepareAbort
+     */
+    EMPTY((byte) 0, org.apache.kafka.clients.admin.TransactionState.EMPTY.toString(), true),
+    /**
+     * Transaction has started and ongoing
+     * <p>
+     * transition: received EndTxnRequest with commit => PrepareCommit
+     *             received EndTxnRequest with abort => PrepareAbort
+     *             received AddPartitionsToTxnRequest => Ongoing
+     *             received AddOffsetsToTxnRequest => Ongoing
+     */
+    ONGOING((byte) 1, org.apache.kafka.clients.admin.TransactionState.ONGOING.toString(), false),
+    /**
+     * Group is preparing to commit
+     * transition: received acks from all partitions => CompleteCommit
+     */
+    PREPARE_COMMIT((byte) 2, org.apache.kafka.clients.admin.TransactionState.PREPARE_COMMIT.toString(), false),
+    /**
+     * Group is preparing to abort
+     * <p>
+     * transition: received acks from all partitions => CompleteAbort
+     * <p>
+     * Note, In transaction v2, we allow Empty, CompleteCommit, CompleteAbort to transition to PrepareAbort. because the
+     * client may not know the txn state on the server side, it needs to send endTxn request when uncertain.
+     */
+    PREPARE_ABORT((byte) 3, org.apache.kafka.clients.admin.TransactionState.PREPARE_ABORT.toString(), false),
+    /**
+     * Group has completed commit
+     * <p>
+     * Will soon be removed from the ongoing transaction cache
+     */
+    COMPLETE_COMMIT((byte) 4, org.apache.kafka.clients.admin.TransactionState.COMPLETE_COMMIT.toString(), true),
+    /**
+     * Group has completed abort
+     * <p>
+     * Will soon be removed from the ongoing transaction cache
+     */
+    COMPLETE_ABORT((byte) 5, org.apache.kafka.clients.admin.TransactionState.COMPLETE_ABORT.toString(), true),
+    /**
+     * TransactionalId has expired and is about to be removed from the transaction cache
+     */
+    DEAD((byte) 6, "Dead", false),
+    /**
+     * We are in the middle of bumping the epoch and fencing out older producers.
+     */
+    PREPARE_EPOCH_FENCE((byte) 7, org.apache.kafka.clients.admin.TransactionState.PREPARE_EPOCH_FENCE.toString(), false);
+
+    private static final Map<String, TransactionState> NAME_TO_ENUM = Arrays.stream(values())
+        .collect(Collectors.toUnmodifiableMap(TransactionState::stateName, Function.identity()));
+
+    private static final Map<Byte, TransactionState> ID_TO_ENUM = Arrays.stream(values())
+        .collect(Collectors.toUnmodifiableMap(TransactionState::id, Function.identity()));
+
+    public static final Set<TransactionState> ALL_STATES = Set.copyOf(EnumSet.allOf(TransactionState.class));
+
+    private final byte id;
+    private final String stateName;
+    public static final Map<TransactionState, Set<TransactionState>> VALID_PREVIOUS_STATES = Map.of(
+        EMPTY, Set.copyOf(EnumSet.of(EMPTY, COMPLETE_COMMIT, COMPLETE_ABORT)),
```

Comment:
Have you consider using `Set.of` instead of `Set.copyOf(EnumSet.of`?

Reply:
Update it. Thanks.